src/Glpi/Controller/DropdownController.php

```diff
@@ -0,0 +1,234 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Controller;
+
+use CommonDevice;
+use CommonDropdown;
+use Html;
+use Glpi\Event;
+use Search;
+use Session;
+use Symfony\Component\HttpFoundation\Exception\BadRequestException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+use Symfony\Component\Routing\Attribute\Route;
+
+final readonly class DropdownController implements Controller
+{
+    public const OPTIONS_KEY = 'dropdown_options';
+
+    #[Route("/dropdown/form/{class}", name: "glpi_dropdown_form", options: ["is_form" => true])]
+    #[Route("/dropdown/{class}", name: "glpi_dropdown")]
```

Comment:
Maybe this should be two separate controllers ?

IMO it does not feel like a single responsibility controller when we have two routes with two different conditioned by a flag argument.

Reply:
Indeed, maybe splitting into a `DropdownIndexController` and a `DropdownFormController` would be better.

Reply:
Done 👌